sycl/include/CL/sycl/INTEL/fpga_reg.hpp

```diff
@@ -14,7 +14,12 @@ __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 namespace INTEL {
 
-template <typename _T> _T fpga_reg(const _T &t) {
+// Returns a registered copy of the input
+// This function is intended for FPGA users to instruct the compiler to insert
+// at least one register stage between the input and the return value.
+template <typename _T>
+typename std::enable_if<std::is_trivially_copyable<_T>::value, _T>::type
```

Comment:
This arguably breaks the API, at least in the sense that some types T that could have previously been used are no longer allowed.  Is the argument that this would never have worked for non-trivially copyable types, and that therefore this isn't really an API breakage?

Reply:
Yes, Kevin - that's correct. In fact, I'd argue that there's a bug here (which this change rectifies) where the hardware doesn't behave correctly. If you pass a non-trivially copyable struct today, in SW, you will get a copy that has its copy constructor called, whereas in HW you're going to get a mem_copy basically.

Reply:
Just passing by...
I do not think using SFINAE will illuminate the end-user.
What about just using a `static_assert` instead with a message explaining the problem and how to solve it?

Reply:
@keryell that's a good point. I tested this out in clang and dpc++ and you get a pretty clean error message. What do you think? 

```
fpga_reg.cpp:77:3: note: in instantiation of function template specialization 'simple_vadd<MyComplex, 4>' requested here
  simple_vadd(deviceQueue, G, H, I);
  ^
./fpga_reg.hpp:23:1: note: candidate template ignored: requirement 'std::is_trivially_copyable<MyComplex>::value' was not satisfied [with _T = MyComplex
```

Reply:
Nice to see already such a nice message with `std::enable_if`! Perhaps it is based nowadays on the equivalent of C++20 concepts...
But there is still some room to make the message more accessible to an FPGA programmer more familiar to C than C++... There is no `template` in the user code and probably `std::is_trivially_copyable<MyComplex>::value' is difficult to understand.

Reply:
I ran your suggestions by a user and they agreed, so I went ahead and switched to the static_assert instead of a type_trait in the function signature instead. @keryell can you review and let me know what you think of the error message, and if it can be made better?